app/routes/_layout+/assets.$assetId.overview.tsx

```diff
@@ -19,64 +23,182 @@ import { Tag } from "~/components/shared/tag";
 import TextualDivider from "~/components/shared/textual-divider";
 import { usePosition } from "~/hooks/use-position";
 import { useUserIsSelfService } from "~/hooks/user-user-is-self-service";
-import type { ASSET_INCLUDE_FIELDS } from "~/modules/asset/fields";
-import { updateAssetBookingAvailability } from "~/modules/asset/service.server";
+import { ASSET_OVERVIEW_FIELDS } from "~/modules/asset/fields";
+import {
+  getAsset,
+  updateAssetBookingAvailability,
+} from "~/modules/asset/service.server";
 import type { ShelfAssetCustomFieldValueType } from "~/modules/asset/types";
+import {
+  createQr,
+  generateCode,
+  getQrByAssetId,
+} from "~/modules/qr/service.server";
+import { getScanByQrId } from "~/modules/scan/service.server";
+import { parseScanData } from "~/modules/scan/utils.server";
 import { appendToMetaTitle } from "~/utils/append-to-meta-title";
 import { checkExhaustiveSwitch } from "~/utils/check-exhaustive-switch";
+import { getDateTimeFormat, getLocale } from "~/utils/client-hints";
 import { getCustomFieldDisplayValue } from "~/utils/custom-fields";
 import { sendNotification } from "~/utils/emitter/send-notification.server";
 import { makeShelfError } from "~/utils/error";
 import { isFormProcessing } from "~/utils/form";
-import { error, getParams, data, parseData } from "~/utils/http.server";
+import {
+  error,
+  getParams,
+  data,
+  parseData,
+  getCurrentSearchParams,
+} from "~/utils/http.server";
 import { isLink } from "~/utils/misc";
+import {
+  PermissionAction,
+  PermissionEntity,
+} from "~/utils/permissions/permission.validator.server";
+import { requirePermission } from "~/utils/roles.server";
 import { tw } from "~/utils/tw";
+
 export const AvailabilityForBookingFormSchema = z.object({
   availableToBook: z
     .string()
     .transform((val) => val === "on")
     .default("false"),
 });
 
-export function loader() {
-  const title = "Asset Overview";
+export async function loader({ context, request, params }: LoaderFunctionArgs) {
+  const authSession = context.getSession();
+  const { userId } = authSession;
+
+  const { assetId: id } = getParams(params, z.object({ assetId: z.string() }), {
+    additionalData: { userId },
+  });
+
+  try {
+    const { organizationId } = await requirePermission({
+      userId,
+      request,
+      entity: PermissionEntity.asset,
+      action: PermissionAction.read,
+    });
+
+    const locale = getLocale(request);
+
+    const asset = await getAsset({
+      id,
+      organizationId,
+      include: ASSET_OVERVIEW_FIELDS,
+    });
+
+    const searchParams = getCurrentSearchParams(request);
+    const size = (searchParams.get("size") || "medium") as SizeKeys;
```

Comment:
We don't need to get this from the search params anymore. Before we used to have a select to choose the size but that is no longer the case so we can skip getting the size from the params.